code/game/objects/items/toys.dm

```diff
@@ -30,21 +30,80 @@
 /*
  * Balloons
  */
-/obj/item/toy/balloon
+
+/obj/item/toy/balloon/
+	var/lastused = null
+
+/obj/item/toy/balloon/attack_self(mob/user)
+	if(world.time - lastused < CLICK_CD_MELEE)
+		return
+	var/playverb = pick("bat [src]", "tug on [src]'s string", "play with [src]")
+	user.visible_message("<span class='notice'>[user] plays with [src].</span>", "<span class='notice'>You [playverb].</span>")
+	lastused = world.time
+
+/obj/item/toy/balloon/syndicateballoon
+	name = "syndicate balloon"
+	desc = "There is a tag on the back that reads \"FUK NT!11!\"."
+	icon_state = "syndballoon"
+	item_state = "syndballoon"
+	w_class = WEIGHT_CLASS_BULKY
+
+
+/*	Coloured balloons, sprite taken from Polaris Station */
+
+/obj/item/toy/balloon/colorballoon
+	name = "balloon"
+	desc = "Colourful fun for all ages! Alien plastic alloys have made it exceptionally hard to pop!"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "colorballoon" /* generic version of this balloon. you can edit its color var to a hex code including # to give it colour, but i wasn't able to get the in hand version to work with changing color vars for some reason */
+	item_state = "colorballoon"
+	w_class = WEIGHT_CLASS_NORMAL
+
+/obj/random/balloon/colorballoon
+	name = "random balloon"
+	desc = "A randomly coloured balloon"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "colorballoonblue"
+
+/obj/random/balloon/colorballoon/item_to_spawn()
+	return pick(subtypesof(/obj/item/toy/balloon/colorballoon))
+
+/obj/item/toy/balloon/colorballoon/colorballoonred
+	icon_state = "colorballoonred"
```

Comment:
You can make a variable on `/obj/random/balloon/colorballoon` level which holds the color.
So something like `var/balloon_color = "blue"`
Then you can use this to set the `icon_state` and `item_state` in the `Initialize`. It can also be used for the description and name.

The instances `/obj/item/toy/balloon/colorballoon/purple` then override this value to their specific color

Reply:
oooh, excellent